pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -57,363 +67,581 @@ def _kokkos_dtype(dtype):
-            ket = ket.astype(np.complex64)
-        else:
-            adj = AdjointJacobianKokkos_C128()
-
-        obs_serialized = _serialize_obs(tape, self.wire_map, use_csingle=self.use_csingle)
-        ops_serialized, use_sp = _serialize_ops(tape, self.wire_map, use_csingle=self.use_csingle)
-
-        ops_serialized = adj.create_ops_list(*ops_serialized)
-
-        trainable_params = sorted(tape.trainable_params)
-
-        tp_shift = []
-        record_tp_rows = []
-        all_params = 0
-
-        for op_idx, tp in enumerate(trainable_params):
-            op, _ = tape.get_operation(
-                op_idx
-            )  # get op_idx-th operator among differentiable operators
-
-            if isinstance(op, Operation) and not isinstance(op, (BasisState, QubitStateVector)):
-                # We now just ignore non-op or state preps
-                tp_shift.append(tp)
-                record_tp_rows.append(all_params)
-            all_params += 1
-
-        if use_sp:
-            # When the first element of the tape is state preparation. Still, I am not sure
-            # whether there must be only one state preparation...
-            tp_shift = [i - 1 for i in tp_shift]
-
-        jac = adj.adjoint_jacobian(self._kokkos_state, obs_serialized, ops_serialized, tp_shift)
-        jac = np.array(jac)  # only for parameters differentiable with the adjoint method
-        jac = jac.reshape(-1, len(tp_shift))
-        jac_r = np.zeros((jac.shape[0], all_params))
-        jac_r[:, record_tp_rows] = jac
-        return jac_r
+            ops_serialized = adj.create_ops_list(*ops_serialized)
+
+            trainable_params = sorted(tape.trainable_params)
+
+            tp_shift = []
+            record_tp_rows = []
+            all_params = 0
+
+            for op_idx, tp in enumerate(trainable_params):
+                op, _ = tape.get_operation(
+                    op_idx
+                )  # get op_idx-th operator among differentiable operators
+
+                if isinstance(op, Operation) and not isinstance(op, (BasisState, QubitStateVector)):
+                    # We now just ignore non-op or state preps
+                    tp_shift.append(tp)
+                    record_tp_rows.append(all_params)
+                all_params += 1
+
+            if use_sp:
+                # When the first element of the tape is state preparation. Still, I am not sure
+                # whether there must be only one state preparation...
+                tp_shift = [i - 1 for i in tp_shift]
+
+            jac = adj.adjoint_jacobian(self._kokkos_state, obs_serialized, ops_serialized, tp_shift)
+            jac = np.array(jac)  # only for parameters differentiable with the adjoint method
+            jac = jac.reshape(-1, len(tp_shift))
+            jac_r = np.zeros((jac.shape[0], all_params))
+            jac_r[:, record_tp_rows] = jac
+            return jac_r
+
+else:  # CPP_BINARY_AVAILABLE:
+
+    class LightningKokkos(QubitDevice):
```

Comment:
I believe we want to fallback to `LightningQubit` here.
```suggestion
    class LightningKokkos(LightningQubit):
```

Reply:
Nice spot! Thanks